status-im/status-react
# Wallet Connect 2.0 integration

fixes #12546 

### Summary

Wallet Connect 2.0 integration. This first approach, as discussed with the team, includes the client library, which is a centralized and experimental approach because it depends on a relay proxy server and the wallet-connect/client package. The reason behind this is that go-waku does not support the waku2 methods that are needed to use to implement a fully decentralized approach, and also status-go does not support having a waku and waku2 connection in parallel. The plan is to remove the `@wallet-connect/client` dependency in the future and replace it with a pure waku2 implementation when feasible.

<img width="448" src="https://user-images.githubusercontent.com/18485527/147983358-37b45356-7a94-4a7b-ba18-ed4edca1654e.png">

<img width="448" src="https://user-images.githubusercontent.com/18485527/147983347-761ab5a5-ddaf-4168-ad46-c2989f6cd29e.png">

<img width="448" src="https://user-images.githubusercontent.com/18485527/147983334-3f189336-b79e-46e7-be4b-adbd6bd82afc.png">

<img width="448" src="https://user-images.githubusercontent.com/18485527/143590750-13833869-7f68-4934-b2a0-c81215802388.png">

<img width="989" src="https://user-images.githubusercontent.com/18485527/143590775-b0c64b29-1a4b-41ea-a1a3-093e48fc6d3f.png">

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

##### Functional

- wallet / transactions
- dapps / app browsing

### Steps to test

- Open Status App
- Go to Advanced settings and enable Wallet Connect 2.0
- Open desktop Dapp, as there are many few amount of dapps using WC 2.0, it can be tested with this example https://react-app.walletconnect.com/
- On desktop Dapp, select chains and click connect
- On Status app, click + on main screen and scan QR
- Wait for the session proposal bottom sheet to appear and connect a wallet
- Execute a transaction in the dapp and wait for signing sheet to appear
- Sign transaction, wait for the result in the dapp
- As a note, `eth_signTypedData` includes a chainId parameter, so wallet should be configured on that same network in order to sign that message
- We should test it with other dapps that implements WC 2.0. Need to research for more besides the testing dapp.

status: ready
